src/main/java/uk/gov/companieshouse/uri/web/transformer/FieldTransformer.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.companieshouse.uri.web.transformer;
+
+public class FieldTransformer {
+    
+    /**
+     * Converts a string date of format 'DD/MM/YYYY' to 'YYYY-MM-DD'
+     * for use in XML template output
+     * If the supplied date is null or is not the correct format, then
+     * the supplied string is returned unaltered
+     * @param displayDateString - date string of the format 'DD/MM/YYYY'
+     */    
+    public String xmlDate(String displayDateString) {
+        if (displayDateString != null && displayDateString.matches("^[0-9]{2}/[0-9]{2}/[0-9]{4}$")) {
```

Comment:
This regex based validation does not guarantee a valid date is formatted but would look like one ..e.g. 29/02/2021 - does that matter - (given other invalid formats are returned as is)?

Reply:
In our case, the date is always going to be valid or null, so that regexp is really just a case of being over cautious.  The date is returned from the api call as a LocalDate and stored as a string for the display model (which is DD/MM/YYYY) as that format is used on all templates except the xml one.

Reply:
using commons-validator GenericValidator.isDate now